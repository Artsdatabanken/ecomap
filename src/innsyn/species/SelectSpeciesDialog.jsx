import React from 'react'
import SelectSpecies from './SelectSpecies'
import Paper from 'material-ui/Paper'

const style2 = {
  padding: 16,
  position: 'absolute',
  left: '1%',
  top: '1%',
  width: '45%',
  height: '98%',
  overflowY: 'auto'
}

const SelectSpeciesDialog = props =>
  <Paper zDepth={4} style={style2}>
    <h2>Legg til kartlag</h2>
    <SelectSpecies />
  </Paper>

export default SelectSpeciesDialog
