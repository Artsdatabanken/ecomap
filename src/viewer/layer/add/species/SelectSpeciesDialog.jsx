import React from 'react'
import propTypes from 'prop-types'
import {FlatButton, Paper} from 'material-ui'
import SelectSpecies from './SelectSpecies'

const style = {
  padding: 16,
  position: 'absolute',
  left: '1%',
  top: '1%',
  width: '98%',
  height: '98%'
}

const SelectSpeciesDialog = ({onAddLayer, onClose}) => {
  return (
    <Paper zDepth={4} style={style}>
      <div style={{ }}>
        <h2>Species</h2>
        <div style={{overflowY: 'auto'}}>
          <SelectSpecies onClick={onAddLayer} />
        </div>
        <FlatButton style={{position: 'fixed', bottom: '24px'}} label='Close' onTouchTap={onClose} />
      </div>
    </Paper>
  )
}

SelectSpeciesDialog.propTypes = {
  onAddLayer: propTypes.func.isRequired,
  onClose: propTypes.func.isRequired
}

export default SelectSpeciesDialog
