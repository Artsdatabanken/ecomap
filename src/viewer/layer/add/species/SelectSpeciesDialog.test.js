import React from 'react'
import renderer from 'react-test-renderer'
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider'

import SelectSpeciesDialog from './SelectSpeciesDialog'

it('renders', () => {
  const tree = renderer.create(
    <MuiThemeProvider>
      <SelectSpeciesDialog onAddLayer={() => {}} onClose={() => {}} />
    </MuiThemeProvider>
      )
  expect(tree.toJSON()).toMatchSnapshot()
})
