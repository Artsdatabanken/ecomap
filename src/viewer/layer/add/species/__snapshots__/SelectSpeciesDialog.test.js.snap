// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 14px 45px rgba(0, 0, 0, 0.25),
         0 10px 18px rgba(0, 0, 0, 0.22)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "height": "98%",
      "left": "1%",
      "muiPrepared": true,
      "padding": 16,
      "position": "absolute",
      "top": "1%",
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
      "width": "98%",
    }
  }
>
  <div
    style={Object {}}
  >
    <h2>
      Species
    </h2>
    <div
      style={
        Object {
          "overflowY": "auto",
        }
      }
    >
      <span>
        <span>
          <svg
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            style={
              Object {
                "color": "rgba(0, 0, 0, 0.87)",
                "display": "inline-block",
                "fill": "currentColor",
                "height": 24,
                "left": 16,
                "muiPrepared": true,
                "position": "relative",
                "top": 8,
                "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                "userSelect": "none",
                "width": 24,
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
          </svg>
          <div
            style={
              Object {
                "display": "inline-block",
                "left": 32,
                "muiPrepared": true,
                "position": "relative",
                "width": "40%",
              }
            }
          >
            <div
              className={undefined}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "cursor": "auto",
                  "display": "inline-block",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 16,
                  "height": 48,
                  "lineHeight": "24px",
                  "muiPrepared": true,
                  "position": "relative",
                  "transition": "height 200ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  "width": 256,
                }
              }
            >
              <div
                style={
                  Object {
                    "bottom": 12,
                    "color": "rgba(0, 0, 0, 0.3)",
                    "muiPrepared": true,
                    "opacity": 1,
                    "position": "absolute",
                    "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                  }
                }
              >
                Search...
              </div>
              <input
                autoComplete="off"
                disabled={false}
                id="sbar"
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                open={false}
                style={
                  Object {
                    "WebkitOpacity": 1,
                    "WebkitTapHighlightColor": "rgba(0,0,0,0)",
                    "appearance": "textfield",
                    "backgroundColor": "rgba(0,0,0,0)",
                    "border": "none",
                    "color": "rgba(0, 0, 0, 0.87)",
                    "cursor": "inherit",
                    "font": "inherit",
                    "height": "100%",
                    "muiPrepared": true,
                    "outline": "none",
                    "padding": 0,
                    "position": "relative",
                    "width": "100%",
                  }
                }
                type="text"
                value=""
              />
            </div>
            <div
              style={
                Object {
                  "display": "none",
                }
              }
            />
          </div>
        </span>
        <div
          style={
            Object {
              "height": 32,
            }
          }
        />
        <span>
          <div
            style={
              Object {
                "columnCount": 4,
              }
            }
          />
        </span>
      </span>
    </div>
    <button
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0)",
          "border": 10,
          "borderRadius": 2,
          "bottom": "24px",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "height": 36,
          "lineHeight": "36px",
          "margin": 0,
          "minWidth": 88,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "hidden",
          "padding": 0,
          "position": "fixed",
          "textAlign": "center",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "userSelect": "none",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <span
          style={
            Object {
              "fontSize": 14,
              "fontWeight": 500,
              "letterSpacing": 0,
              "muiPrepared": true,
              "paddingLeft": 16,
              "paddingRight": 16,
              "position": "relative",
              "textTransform": "uppercase",
              "verticalAlign": "middle",
            }
          }
        >
          Close
        </span>
      </div>
    </button>
  </div>
</div>
`;
